### PR TITLE
pyDoc2GitHub CWE-191

### PR DESCRIPTION
**Pull request by myteron**
Adding code and doc for CWE-191 as part of #531 .

### CODE REVIEW

File: docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-191/README.md [hunk 1]
```diff
@@ -1,3 +1,183 @@
 # CWE-191, Integer Underflow (Wrap or Wraparound)
 
-PLACEHOLDER for # CWE-1335: Promote readability and compatibility by using mathematical written code with arithmetic operations instead of bit-wise operations link
+Ensure that integer overflow is properly handled in order to avoid unexpected behavior. Python data types can be divided into two categories:
+
+- Built-in types such as `int`, `float`, or `complex` [[Python 2024]](https://docs.python.org/3.9/library/stdtypes.html). These types are provided by classes and are protected against overflows.
+
+- Primitive types such as `numpy`, `time`, or `ctype`  share issues known from `C` , or `C++`  and are often used to interact with the operating system, or for efficiency. Developers should follow the `C` guidelines for rules to follow.
```
Comment (gkunz) [inline docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-191/README.md:7]: There is a word missing: "... interact with the operating system, or (?) for efficiency..."

File: docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-191/README.md [hunk 2]
```diff
@@ -1,3 +1,183 @@
...
+
+## Non-Compliant Code Example
+
+Using a `numpy.int64` can cause an unintentional flip of its sign when reaching the maximum number that can be store as demonstrated in `noncompliant01.py`.
```
Comment (gkunz) [inline docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-191/README.md:11]:
```suggestion
Using a `numpy.int64` can cause an unintentional flip of its sign when reaching the maximum number that can be stored as demonstrated in `noncompliant01.py`.
```

File: docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-191/README.md [hunk 3]
```diff
@@ -1,3 +1,183 @@
...
+import numpy
+
+a = numpy.int64(numpy.iinfo(numpy.int64).max)
+print(a + 1)  # RuntimeWarning and continues
+print()
+b = numpy.int64(numpy.iinfo(numpy.int64).max + 1)  # OverflowError and stops
+print(b)  # we will never reach this
+```
+
+Adding `+1` to `9223372036854775807` results in a negative number `-9223372036854775808` throws a `RuntimeWarning`  but continues.
```
Comment (gkunz) [inline docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-191/README.md:27]:
```suggestion
Adding `+1` to `9223372036854775807` results in a negative number `-9223372036854775808` and throws a `RuntimeWarning`  but continues.
```

File: docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-191/README.md [hunk 4]
```diff
@@ -1,3 +1,183 @@
...
+
+An attempt to create `int` from a too big number causes an `OverflowError` and stops.
+
+> [!NOTE]
+> It has been observed that different result may occur depending on `numpy` version for reference we are using `numpy 1.23.1` and Python version: `3.9.12.`
```
Comment (gkunz) [inline docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-191/README.md:32]:
```suggestion
> It has been observed that different results may occur depending on the version of `numpy`. For reference, we are using `numpy 1.23.1` and Python `3.9.12.`
```

File: docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-191/README.md [hunk 5]
```diff
@@ -1,3 +1,183 @@
...
+ 
+#####################
+# exploiting above code example
+#####################
+print(get_time_in_future(23**74))
+```
+
+## Compliant Solution
+
+This `compliant02.py` solution handles `OverflowError` Exception when a to high value is given to `get_time_in_future`.
```
Comment (gkunz) [inline docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-191/README.md:90]:
```suggestion
This `compliant02.py` solution handles `OverflowError` Exception when a too large value is given to `get_time_in_future`.
```

File: docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-191/README.md (hunk 1 above)
Comment (myteron) [reply docs/Secure-Coding-Guide-for-Python/CWE-682/CWE-191/README.md]: changed as discussed

**Review by s19110 — APPROVED** [review]
The README page looks good and the code examples work as intended.

**Review by gkunz — APPROVED** [review]

**Pull request merged**